src/systems/personal-space-bubble.js

```diff
@@ -234,10 +234,36 @@ AFRAME.registerComponent("personal-space-invader", {
   applyInvasionToMesh(invading) {
     if (this.disabled) return;
 
-    if (this.targetMesh && this.targetMesh.material && !this.alwaysHidden) {
-      forEachMaterial(this.targetMesh, material => {
-        material.opacity = invading ? this.data.invadingOpacity : 1;
-        material.transparent = invading;
+    // Note: Model isn't loaded on init because this object is inflated before the root is.
+    // object3DMap.mesh is not initially set and must be checked before traversing.
+    if (this.gltfRootEl && this.gltfRootEl.object3DMap.mesh && !this.alwaysHidden) {
+      traverseFilteredSubtrees(this.gltfRootEl.object3DMap.mesh, obj => {
+        // Prevents changing the opacity of ui elements
+        if (obj.el && obj.el.components.tags && obj.el.components.tags.data.ignoreSpaceBubble) {
+          // Skip all objects under this branch by returning false
+          return false;
+        }
+
+        if (!obj.material) {
+          return;
+        }
+
+        forEachMaterial(obj, material => {
+          let originalProps = material.userData.originalProps;
+
+          if (!material.userData.originalProps) {
+            originalProps = material.userData.originalProps = {
+              opacity: material.opacity,
+              transparent: material.transparent
+            };
+          }
+
+          // Note: Sharing materials will cause all objects with the material to turn transparent / opaque
+          // This is generally fine for avatars since a material will only be shared within a glTF model,
+          // not across avatars in the room.
+          material.opacity = invading ? this.data.invadingOpacity : originalProps.opacity;
+          material.transparent = invading ? true : originalProps.transparent;
```

Comment:
This could be written as `material.transparent = invading || originalProps.transparent`. It's a tossup which one actually reads more clearly, but a ternary with `true` in it stood out so figured I would mention it.